pkg/network/namescheme/networknamescheme.go

```diff
@@ -91,3 +96,51 @@ func secondaryInterfaceIndexedName(idx int) string {
 	const interfaceNamePrefix = "net"
 	return fmt.Sprintf("%s%d", interfaceNamePrefix, idx)
 }
+
+// CreateNetworkNameSchemeByPodNetworkStatus create pod network name scheme according to the given VMI spec networks
+// and pod annotation.
+// In case Multus network-status annotation exist and reflect ordinal interface names it returns an ordinal network
+// name scheme, Otherwise, returns the hashed network name scheme.
+func CreateNetworkNameSchemeByPodNetworkStatus(networks []v1.Network, podAnnotations map[string]string) map[string]string {
+	networksToPodIface := CreateNetworkNameScheme(networks)
+
+	podNetworkStatusAnnotation, found := podAnnotations[networkv1.NetworkStatusAnnot]
```

Comment:
`NetworkStatusAnnot`is the reported information which multus provides, while `NetworkAttachmentAnnot` is the one specified by the client (the desired).

Why do we need to depend on the status and not on the spec?
I commented below about this being potentially unsafe.

Reply:
Ok, I hope this will not bite is in the future.